src/identity/utils/getIdentityThunk.ts

```diff
@@ -0,0 +1,10 @@
+import {CLOUD} from 'src/shared/constants'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {getQuartzMeThunk} from 'src/me/actions/thunks'
+import {getQuartzIdentityThunk} from '../actions/thunks'
+
+// For now, check for both uiUnificationFlag and quartzIdentity to ensure this isn't used in tools.
+export const getIdentityThunk = () =>
+  CLOUD && isFlagEnabled('uiUnificationFlag') && isFlagEnabled('quartzIdentity')
+    ? getQuartzIdentityThunk()
+    : getQuartzMeThunk()
```

Comment:
100% right that `CLOUD` is a precondition for both, so I don't think the utility was working as I expected. As far as moving it into the thunk itself, that makes sense to me, here's what I've changed:

Everything that previously used the `getIdentityThunk` utility to decide between `getQuartzMe` and `getQuartzIdentity` thunks now just does one thing - calls `getQuartzIdentity`, which has this logic:

Return / do nothing if we're not in an environment with quartz. Still confirming this behaves as expected since we'd be returning out of the thunk without dispatching any actions.
`if (!CLOUD || !isFlagEnabled('uiUnificationFlag')) {
    return
 }`

If the `quartzIdentity` feature flag is off, just use the legacy thunk.
`
  if (!isFlagEnabled('quartzIdentity')) {
    return getQuartzMeThunk()
 }`

The rest of the logic is the same `getQuartzIdentityThunk`. I like the improvement - requires less cleanup after `quartz/me` is deprecated.

Also implemented the `shouldUseQuartzIdentity` idea, which is great and saves us some lines throughout.